### PR TITLE
Add a script to cycle through language switching modes.

### LINKED ISSUE #10253: Add a script to toggle language detection
### Is your feature request related to a problem? Please describe.
In some cases, I need to deactivate auto-language switching:

* Sometimes webpages are tagged with wrong language, e.g. written in French but tagged as English content. Probably due to a webmaster using a template or a default setting in English.
* Sometimes I prefer reading English-tagged content with my default voice native language (French), e.g. math content.

On the opposite, sometimes I prefer have auto-language switching on to get used to real language pronunciation.
Pressing NVDA+Ctrl+V to go to voice settings panel, then tabbing in the dialog box to go to auto language switching option, activate it and press Enter is a bit long.

### Describe the solution you'd like
Add a script to toggle automatic language switching. Such scripts exist for many other setting panels options, so this would be coherent. The script may be unassigned and could be assigned by the user as already done for many other such setting toggle scripts.
The script could also include auto dialect switching option and toggle between 3 states:

* Language and dialect detection off
* Language detection on, dialect off
* Language and dialect detection on

Note that the 4th possible state (language detection off and dialect on) does not make sense to me.

### Describe alternatives you've considered
* Alternative 1: Add an accelerator on automatic language switching option in the speech setting panel. Even if it allows accelerating a bit the process, it is not as fast as a dedicated script.
* Alternative 2: Toggle only auto language switching option, not auto dialect switching.

### Additional context
I have a personal script that does the job here:
https://raw.githubusercontent.com/CyrilleB79/NVDAScripts/master/globalPlugins/autoLangSwitch.py

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:

Fixes #10253

### Summary of the issue:

A gesture is missing to control automatic language switching and automatic dialect switching options.

### Description of how this pull request fixes the issue:
Add an unassigned script that cycles through the 3 following automatic language and dialect switching modes:
* Language and dialect switching off
* Language switching on, dialect switching off
• Language and dialect switching on

Note that the 4th possible state (language switching off and dialect switching on) does not make sense.

### Testing strategy:

* Called the script many times and checked the result in the Speech Settings panel after each call
* Tested with eSpeak in French (Franche) the text on this test page: [ExamplePageWithLangTag.html.txt](https://github.com/nvaccess/nvda/files/8665023/ExamplePageWithLangTag.html.txt)
  Note: The number 70 is said to be "soixante-dix" in French from France and "septante" in French from Belgium. Hence the easy test to distinguish one dialect from another.

 ### Known issues with pull request:

In #10253, the alternative solution to include this toggle in the synth setting ring was suggested by @JulienCochuyt.
Since the synth settings ring already contains many parameters, I have not retained this solution. Should #7747 be implemented, we can re-discuss it at that time.
Also a dedicated gesture is faster than the ring and thus would fulfill the expectations of https://github.com/nvaccess/nvda/issues/10253#issuecomment-1122387480.

### Change log entries:
New features
`An unassigned script to cycle through language and dialect switching modes have been added. (#10253)`

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/globalCommands.py [hunk 1]
```diff
@@ -898,6 +898,31 @@ def script_toggleReportClickable(self,gesture):
 			config.conf["documentFormatting"]["reportClickable"]=True
 		ui.message(state)
 
+	@script(
+		description=_(
+			# Translators: Input help mode message for cycle through automatic language switching mode command.
+			"Cycles through speech automatic language switching modes off, language only and language and dialect."
```
Comment (seanbudd) [inline source/globalCommands.py:904]: I found this sentence hard to parse. Alternative suggestion (adds one word):

```suggestion
			"Cycles through speech modes for automatic language switching: off, language only and language and dialect."
```

**Comment by seanbudd** [comment]
Thanks @CyrilleB79  - LGTM otherwise

File: source/globalCommands.py (hunk 1 above)
Comment (CyrilleB79) [reply source/globalCommands.py]: Thanks. Fixed in cfa51bafe7a460066828d19c0648401c3b1bca21.

**Review by seanbudd — APPROVED** [review]
